ggshield/cmd/iac/scan.py

```diff
@@ -46,15 +61,98 @@ def validate_exclude(_ctx: Any, _param: Any, value: Sequence[str]) -> None:
 )
 @click.option("--json", is_flag=True, help="JSON output.")
 @click.argument(
-    "directory", type=click.Path(exists=True, readable=True, path_type=type(Path))
+    "directory", type=click.Path(exists=True, readable=True, path_type=Path)
 )
+@click.pass_context
 def scan_cmd(
+    ctx: click.Context,
     exit_zero: bool,
-    level: str,
+    minimum_severity: str,
     verbose: bool,
     ignore_policies: Sequence[str],
     ignore_paths: Sequence[str],
     json: bool,
     directory: Path,
+) -> int:
+    update_context(
+        ctx, exit_zero, minimum_severity, verbose, ignore_policies, ignore_paths, json
+    )
+    result = iac_scan(ctx, directory)
+    scan = ScanCollection(id=str(directory), type="path_scan", iac_result=result)
+
+    output_handler: OutputHandler = ctx.obj["output_handler"]
+    return output_handler.process_scan(scan)
+
+
+def update_context(
+    ctx: click.Context,
+    exit_zero: bool,
+    minimum_severity: str,
+    verbose: bool,
+    ignore_policies: Sequence[str],
+    ignore_paths: Sequence[str],
+    json: bool,
 ) -> None:
-    pass
+    config: Config = ctx.obj["config"]
+    ctx.obj["client"] = create_client_from_config(config)
+
+    ignored_paths = config.user_config.iac.ignored_paths
+    if ignore_paths is not None:
+        ignored_paths.update(ignore_paths)
+
+    ctx.obj["exclusion_regexes"] = init_exclusion_regexes(ignored_paths)
+
+    ignored_policies = config.user_config.iac.ignored_policies
+    if ignore_policies is not None:
+        ignored_policies.update(ignore_policies)
```

Comment:
```suggestion
    if ignore_policies is not None:
        config.user_config.iac.ignored_policies.update(ignore_policies)
```

Would make it easier to understand what is going on (`ignore_policies` and `ignored_policies` are difficult to distinguish). Same for `ignored_paths`.